crates/fluvio/src/producer/partition_producer.rs

```diff
@@ -213,23 +245,83 @@ impl PartitionProducer {
         request.isolation = self.config.isolation;
         request.timeout = self.config.timeout;
         request.topics.push(topic_request);
-        let response = spu_socket.send_receive(request).await?;
+
+        let (response, stats_update) = if self.client_stats.is_some() {
+            Self::measure_send_receive(&spu_socket, request).await?
```

Comment:
This would be more cleaner like this:
```
if let Some(stat) = self.client.stats {
     stat.measure_send_receive(&spu_socket,request).await?
}
```